include/cynq/memory.hpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * See LICENSE for more information about licensing
+ *
+ * Copyright 2023
+ * Author: Luis G. Leon-Vega <luis.leon@ieee.org>
+ *         Diego Arturo Avila Torres <diego.avila@uned.cr>
+ *
+ */
+#pragma once
+#include <memory>
+
+#include "cynq/enums.hpp"
+#include "cynq/status.hpp"
+
+namespace cynq {
+class IMemory {
+ public:
+  virtual ~IMemory() {}
+  /**
+   * @brief Sync method
+   * Synchronizes the memory.
```

Comment:
Be more explicit. The synchronisation types are from host to device to vice-versa. Be more explicit